File: javascript-closure-angular-client/API/Client/TaxByTypeSummaryForGoods.js
goog.provide('API.Client.TaxByTypeSummaryForGoods');

/**
 * summary of all taxes
 * @record
 */
API.Client.TaxByTypeSummaryForGoods = function() {}

/**
 * sum of all lines calcbase
 * @type {!number}
 * @export
 */
API.Client.TaxByTypeSummaryForGoods.prototype.calcbase;

/**
 * sum of referenced tax value
 * @type {!number}
 * @export
 */
API.Client.TaxByTypeSummaryForGoods.prototype.tax;

/**
 * @type {!Array<!API.Client.TaxByTypeSummaryJurisdictionForGoods>}
 * @export
 */
API.Client.TaxByTypeSummaryForGoods.prototype.jurisdictions;

File: CsharpDotNet2-client/src/main/CsharpDotNet2/IO/Swagger/Model/TaxByTypeSummaryForGoods.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// summary of all taxes
  /// </summary>
  [DataContract]
  public class TaxByTypeSummaryForGoods {
    /// <summary>
    /// sum of all lines calcbase
    /// </summary>
    /// <value>sum of all lines calcbase</value>
    [DataMember(Name="calcbase", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "calcbase")]
    public double? Calcbase { get; set; }

    /// <summary>
    /// sum of referenced tax value
    /// </summary>
    /// <value>sum of referenced tax value</value>
    [DataMember(Name="tax", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "tax")]
    public double? Tax { get; set; }

    /// <summary>
    /// Gets or Sets Jurisdictions
    /// </summary>
    [DataMember(Name="jurisdictions", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "jurisdictions")]
    public List<TaxByTypeSummaryJurisdictionForGoods> Jurisdictions { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class TaxByTypeSummaryForGoods {\n");
      sb.Append("  Calcbase: ").Append(Calcbase).Append("\n");
      sb.Append("  Tax: ").Append(Tax).Append("\n");
      sb.Append("  Jurisdictions: ").Append(Jurisdictions).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}
